racetrack_client/racetrack_client/client/deploy.py

```diff
@@ -116,7 +116,8 @@ def send_deploy_request(
             job_url = _wait_for_deployment_result(lifecycle_url, deploy_id, user_auth, [])
         except Exception as e:
             raise DeploymentError(e)
-
+        if warnings := _get_warning(lifecycle_url, deploy_id, user_auth):
```

Comment:
This looks unnecessary, you get warnings field already while getting deployment data JSON in `_wait_for_deployment_result`. So another request is redundant.